### ISSUE TITLE
Yarn integrity errors breaking build

### ISSUE THREAD

**Issue by pun-ky** [node-gradle/gradle-node-plugin#49]
On some dev machines, under totally strange circumstances this kicks builds using node-gradle/gradle-node-plugin :

https://github.com/yarnpkg/yarn/issues/7584

current workaround for this is to run `sh gradlew yarn_cache_clean`

maybe sth else could be done like upgrading default version of yarn (hopefully if it was fixed in the meantime) to prevent this issue occuring in case of using this plugin?

**Comment by pun-ky** [comment]
just upgrade to default yarn version to 1.19.1 will be very appreciated

**Comment by deepy** [comment]
Given that 1.19.1 is the current stable that should be fine

**Comment by robclancy** [comment]
I'm on 1.19.2 and this still happens.

**Comment by pun-ky** [comment]
seems that yarn cache sometimes is reusable but need to be upgraded / migrated and looks like that that kind of migration may fail and end with integrity error. As I understand that, it's rather one timer and build just fails when some older version of yarn was launched before. and that cache need to be deleted. I am not sure if there will be any fix introduced that will remove a need for that cache deletion.

**Comment by robclancy** [comment]
I have to clean the cache every single time. Anything changes at all I need to do `yarn cache clean` and then `yarn install -f --update-checksums`. Super annoying.

**Comment by pun-ky** [comment]
agree, so seems that this need to be addressed somehow / more elegantly
